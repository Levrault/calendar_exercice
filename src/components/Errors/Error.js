import React from 'react';
import PropTypes from 'prop-types';
import PageNotFound from './PageNotFound';
import styles from './Error.css';

const register = {
  404: PageNotFound
};

/**
 * Show message error
 * @param {string} message
 */
const Error = ({ response }) => {
  const Component = register[response.status];
  if (!Component) return null;
  return (
    <div className={styles.container} >
      <Component />
    </div>
  );
};

Error.propTypes = {
  response: PropTypes.object.isRequired
};

export default Error;
