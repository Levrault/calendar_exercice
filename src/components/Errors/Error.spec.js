import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from '../../tools/test/enzyme';
import Error from './Error';

describe('Error component', () => {
  it('should render without problem', () => {
    const component = renderer.create(
      <Error response={{ status: 404 }} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return null if error isn\'t mapped', () => {
    const component = shallow(
      <Error response={{ status: -1 }} />,
    );
    expect(component.type()).toBeNull();
  });
});
