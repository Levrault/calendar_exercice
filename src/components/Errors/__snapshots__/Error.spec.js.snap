// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should render without problem 1`] = `
<div>
  <div>
    404 not found
  </div>
</div>
`;
